pilot/cmd/sidecar-injector/main.go

```diff
@@ -109,26 +111,48 @@ var (
 	}
 )
 
-func patchCert() error {
-	const retryTimes = 6 // Try for one minute.
+// patchCertLoop continually reapplies the caBundle PEM. This is required because it can be overwritten with empty
+// values if the original yaml is reapplied (https://github.com/istio/istio/issues/6069).
+func patchCertLoop() error {
 	client, err := createClientset(flags.kubeconfigFile)
 	if err != nil {
 		return err
 	}
+
 	caCertPem, err := ioutil.ReadFile(flags.caCertFile)
 	if err != nil {
 		return err
 	}
-	for i := 0; i < retryTimes; i++ {
-		err = util.PatchMutatingWebhookConfig(client.AdmissionregistrationV1beta1().MutatingWebhookConfigurations(),
-			flags.webhookConfigName, flags.webhookName, caCertPem)
-		if err == nil {
-			return nil
+
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		return err
+	}
+
+	watchDir, _ := filepath.Split(flags.caCertFile)
+	if err := watcher.Watch(watchDir); err != nil {
+		return fmt.Errorf("could not watch %v: %v", flags.caCertFile, err)
+	}
+
+	tickerC := time.NewTicker(time.Second).C
+	for {
+		select {
+		case <-tickerC:
+			err = util.PatchMutatingWebhookConfig(client.AdmissionregistrationV1beta1().MutatingWebhookConfigurations(),
+				flags.webhookConfigName, flags.webhookName, caCertPem)
+			if err == nil {
+				log.Errorf("Patch webhook failed: %s", err)
+			}
+
+		case <-watcher.Event:
+			caCertPem, err = ioutil.ReadFile(flags.caCertFile)
```

Comment:
This doesn't filter on file event type (e.g. delete). Any chance that this read will fail in such cases and overwrite `caCertPem` with nil? Maybe something like the following?

```golang
if b, err := ioutil.ReadFile(flags.caCertFile); err != nil { 
    /* error */ 
} else {
    caCertPem = b
}
```

Reply:
Done